apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -217,17 +217,26 @@ namespace MWRender
         , mFieldOfViewOverride(0.f)
         , mBorders(false)
     {
-        resourceSystem->getSceneManager()->setParticleSystemMask(MWRender::Mask_ParticleSystem);
-        resourceSystem->getSceneManager()->setShaderPath(resourcePath + "/shaders");
-        resourceSystem->getSceneManager()->setForceShaders(Settings::Manager::getBool("force shaders", "Shaders"));
-        resourceSystem->getSceneManager()->setClampLighting(Settings::Manager::getBool("clamp lighting", "Shaders"));
-        resourceSystem->getSceneManager()->setForcePerPixelLighting(Settings::Manager::getBool("force per pixel lighting", "Shaders"));
-        resourceSystem->getSceneManager()->setAccurateFog(Settings::Manager::getBool("accurate fog", "Shaders"));
-        resourceSystem->getSceneManager()->setAutoUseNormalMaps(Settings::Manager::getBool("auto use object normal maps", "Shaders"));
-        resourceSystem->getSceneManager()->setNormalMapPattern(Settings::Manager::getString("normal map pattern", "Shaders"));
-        resourceSystem->getSceneManager()->setNormalHeightMapPattern(Settings::Manager::getString("normal height map pattern", "Shaders"));
-        resourceSystem->getSceneManager()->setAutoUseSpecularMaps(Settings::Manager::getBool("auto use object specular maps", "Shaders"));
-        resourceSystem->getSceneManager()->setSpecularMapPattern(Settings::Manager::getString("specular map pattern", "Shaders"));
+        {
+            Resource::SceneManager& s = *resourceSystem->getSceneManager();
+
+            s.setParticleSystemMask(MWRender::Mask_ParticleSystem);
+            s.setShaderPath(resourcePath + "/shaders");
+            s.setForceShaders(Settings::Manager::getBool("force shaders", "Shaders"));
+            s.setClampLighting(Settings::Manager::getBool("clamp lighting", "Shaders"));
+            s.setForcePerPixelLighting(Settings::Manager::getBool("force per pixel lighting", "Shaders"));
+            s.setAccurateFog(Settings::Manager::getBool("accurate fog", "Fog"));
+            s.setAutoUseNormalMaps(Settings::Manager::getBool("auto use object normal maps", "Shaders"));
+            s.setNormalMapPattern(Settings::Manager::getString("normal map pattern", "Shaders"));
+            s.setNormalHeightMapPattern(Settings::Manager::getString("normal height map pattern", "Shaders"));
+            s.setAutoUseSpecularMaps(Settings::Manager::getBool("auto use object specular maps", "Shaders"));
+            s.setSpecularMapPattern(Settings::Manager::getString("specular map pattern", "Shaders"));
+
+            if (s.getClampLighting())
+                s.setForceShaders(true);
+            if (s.getAccurateFog())
+                s.setForceShaders(true);
+        }
```

Comment:
Doing stuff like this isn't a good idea on big projects with lots of people. It might be a bit tidier now, but you've changed a bunch of stuff that isn't in the scope of this PR, made it harder to use git blame to work out where things came from and made merge conflicts with other stuff a lot more likely.

Also, `s` is not a good variable name, even for a context as small as this. `sceneManager` would have been much better.

Reply:
Hey, but this *is* self-documenting regardless of variable name length.